src/NuGet.Core/NuGet.Packaging/Signing/Verification/AllowListVerificationProvider.cs

```diff
@@ -106,6 +106,11 @@ private Task<SignedPackageVerificationResult> VerifyAllowList(CertificateListVer
                         if (IsSignatureTargeted(certificateHashEntry.Target, signature) &&
                             StringComparer.OrdinalIgnoreCase.Equals(certificateHashEntry.Fingerprint, primarySignatureCertificateFingerprint))
                         {
+                            if (ShouldVerifyOwners(certificateHashEntry as TrustedRepositoryAllowListEntry, signature as IRepositorySignature, out var allowedOwners, out var actualOwners))
```

Comment:
This will never throw since if it's not of that type it will return `null`. Do you think it would be more readable to have the code inline instead of a separate function and have it repeated in both places it needed to avoid the multiple `out` arguments?